src/uri/uri.c

```diff
@@ -271,3 +273,75 @@ int uri_headers_apply(const struct pl *pl, uri_apply_h *ah, void *arg)
 
 	return err;
 }
+
+
+/**
+ * Auto complete a SIP uri, add scheme and domain if missing
+ *
+ * @param uri Input SIP uri
+ * @param hostport host[:port] of SIP URI
+ * @param buf Target buffer to print SIP uri
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int uri_complete(const char *uri, const char *hostport, struct mbuf *buf)
```

Comment:
please note that `struct uri` is generic URI code

this new function is specific to SIP only.

I think this function is too specific to belong in a general purpose library.
It is better to be located in baresip, which is also a library -> libbaresip.so